File: Ruben - Modulo 3/MyRobot/app/src/main/java/com/yezarougmail/myrobot/MainActivity.java
package com.yezarougmail.myrobot;

import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    TextView tv;
    RadioButton rbb;
    RadioButton rbg;
    RadioGroup rg;
    ImageView iv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tv = findViewById(R.id.textView);
        rbb = findViewById(R.id.radioButtonBlue);
        rbg = findViewById(R.id.radioButtonGreen);
        rg = (RadioGroup) findViewById(R.id.radioGroup);
        iv = findViewById(R.id.imageView);
    }

    public void botonPulsado(View v){
        tv.setBackgroundColor(Color.RED);
        tv.setTextColor(Color.WHITE);
    }

    public void robotPulsado(View v){
        iv.setImageResource(R.mipmap.ic_launcher_round);
        switch (rg.getCheckedRadioButtonId()){
            case R.id.radioButtonBlue: tv.setBackgroundColor(0xFF0000FF); tv.setTextColor(Color.WHITE); break;
            case R.id.radioButtonGreen: tv.setBackgroundColor(0xFF00FF00); break;
        }

//        if (rbb.isChecked()) tv.setBackgroundColor(0xFF0000FF);
//        else if (rbg.isChecked()) tv.setBackgroundColor(0xFF00FF00);
    }
}

File: Ruben - Modulo 3/RobotinKotlin/app/src/main/java/com/example/android9ed/robotinkotlin/MainActivity.kt
package com.example.android9ed.robotinkotlin

import android.graphics.Color
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.os.PersistableBundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var color: Int = 0xFFFFFFFF.toInt()
    var imageChanged: Boolean = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val myListener = myL()

        if (savedInstanceState != null) {
            color = savedInstanceState.getInt("Color", color)
            imageChanged = savedInstanceState.getBoolean("imgChanged", false)
            if (imageChanged)imageView.setImageResource(R.mipmap.ic_launcher_round)
            textView.setBackgroundColor(color)
        }

        buttonRed.setOnClickListener(myListener)
        imageView.setOnClickListener(myListener)
    }


    override fun onSaveInstanceState(outState: Bundle?) {
        super.onSaveInstanceState(outState)
        outState?.putInt("color", color)
        outState?.putBoolean("imgChanged", imageChanged)
    }

    private inner class myL: View.OnClickListener
    {
        override fun onClick(v: View?){
            when (v){
                buttonRed -> {
                    if (editText.text.toString().matches("0x[a-fA-F0-9]{8}".toRegex())) {
                        var longcolor: Long = editText.text.substring(2).toLong(16)
                        color = longcolor.toInt()
                    } else color = Color.RED
                }
                imageView -> {
                    if (radioButtonBlue.isChecked)
                        color = (Color.BLUE)
                    else if (radioButtonGreen.isChecked)
                        color = Color.GREEN

                    imageView.setImageResource(R.mipmap.ic_launcher_round)
                }
            }

            textView.setBackgroundColor(color)
        }
    }
}
